sdk/tests/conformance2/renderbuffers/multisampled-depth-renderbuffer-initialization.html

```diff
@@ -0,0 +1,172 @@
+}
+
+function runTest(gl, params) {
+    debug("Test for depth buffer: " + JSON.stringify(params));
+    let resolve = params.alloc2 ? params.alloc2 : params.alloc1;
+    gl.viewport(0, 0, resolve.w, resolve.h);
+    wtu.checkCanvasRect(gl, 0, 0, resolve.w, resolve.h,
+                        [0, 0, 0, 0],
+                        "internal buffers have been initialized to 0");
+
+    gl.disable(gl.DEPTH_TEST);
+
+    // fill the back buffer so we know that reading below happens from
+    // the renderbuffer.
+    gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+
+    // Set up (color+depth) non-multisampled buffer to blit to and read back from.
+    var fbo = gl.createFramebuffer();
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+    var buffer = gl.createRenderbuffer();
+    gl.bindRenderbuffer(gl.RENDERBUFFER, buffer);
+    gl.renderbufferStorage(gl.RENDERBUFFER, gl.RGBA8, resolve.w, resolve.h);
+    gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.RENDERBUFFER, buffer);
+    var depthBuffer = gl.createRenderbuffer();
+    gl.bindRenderbuffer(gl.RENDERBUFFER, depthBuffer);
+    gl.renderbufferStorage(gl.RENDERBUFFER, gl.DEPTH_COMPONENT16, resolve.w, resolve.h);
+    gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.DEPTH_ATTACHMENT, gl.RENDERBUFFER, depthBuffer);
+    shouldBe("gl.checkFramebufferStatus(gl.FRAMEBUFFER)",
+             "gl.FRAMEBUFFER_COMPLETE");
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, 'should be no errors');
+    gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+    wtu.checkCanvasRect(gl, 0, 0, resolve.w, resolve.h,
+                        [0, 255, 0, 255],
+                        "user buffer has been cleared to green");
+
+    // Set up (depth-only) multisampled buffer to test.
+    var fbo_m = gl.createFramebuffer();
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fbo_m);
+    var buffer_m = gl.createRenderbuffer();
+    gl.bindRenderbuffer(gl.RENDERBUFFER, buffer_m);
+
+    if (params.alloc1) {
+        allocStorage(params.alloc1.w, params.alloc1.h, params.alloc1.s);
+    }
+    gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.DEPTH_ATTACHMENT, gl.RENDERBUFFER, buffer_m);
+    if (params.alloc2) {
+        if (params.alloc1) {
+            // Clear the FBO in order to make sure framebufferRenderbuffer is
+            // committed. (In Firefox, framebufferRenderbuffer is deferred, so this
+            // is needed to trigger the bug.)
+            gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+        }
+        allocStorage(params.alloc2.w, params.alloc2.h, params.alloc2.s);
+    }
+
+    function allocStorage(width, height, samples) {
+        gl.renderbufferStorageMultisample(
+            gl.RENDERBUFFER, samples, gl.DEPTH_COMPONENT16, width, height);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+            "should be no error after renderbufferStorageMultisample(DEPTH_COMPONENT16).");
+    }
+
+    shouldBe("gl.checkFramebufferStatus(gl.FRAMEBUFFER)",
+             "gl.FRAMEBUFFER_COMPLETE");
```

Comment:
Let's return early if this is not FRAMEBUFFER_COMPLETE rather than failing the test.  On desktop GL, FRAMEBUFFER_COMPLETE isn't guaranteed, especially on a depth image only situation.